eng/scripts/vcpkg.ps1

```diff
@@ -0,0 +1,52 @@
+[CmdletBinding()]
+Param (
+    [Parameter()]
+    [ValidateNotNullOrEmpty()]
+    [string] $Ref = 'master',
+
+    [Parameter(Mandatory = $true)]
+    [ValidateNotNullOrEmpty()]
+    [string] $Dependencies,
+
+    [Parameter()]
+    [string] $TargetPath ="$env:TEMP/$([guid]::NewGuid())",
+
+    [Parameter()]
+    [switch] $DevOpsLogging = ($null -ne $env:SYSTEM_TEAMPROJECTID)
+)
+
+function SetEnvironmentVariable {
+    param(
+        [string] $Name,
+        [string] $Value
+    )
+
+    if ($devOpsLogging) {
+        Write-Host "##vso[task.setvariable variable=$Name]$($Value)"
+    } else {
+        Write-Verbose "Setting local environment variable: $Name = ***"
+        Set-Item -Path "env:$Name" -Value $Value
+    }
+}
+
+try {
+    New-Item -ItemType Directory -Path $TargetPath -Force
+    Push-Location $TargetPath
+
+    git clone https://github.com/Microsoft/vcpkg .
+    git checkout (Get-Content "$PSScriptRoot/../vcpkg-ref.txt")
```

Comment:
Fetch the tags to avoid silent failure on checkout and log these steps to console. 

See https://github.com/Azure/azure-sdk-for-c/pull/1143#discussion_r478115699